roofit/hs3/src/HistFactoryJSONTool.cxx

```diff
@@ -0,0 +1,255 @@
+   simtags.set_seq();
+   simtags.append_child() << "toplevel";
+   auto &ch = sim["channels"];
+   ch.set_map();
+   for (const auto &c : this->_measurement->GetChannels()) {
+      auto &thisch = ch[c.GetName()];
+      this->Export(c, thisch);
+   }
+
+   // the variables
+   auto &varlist = n["variables"];
+   varlist.set_map();
+   for (const auto &c : this->_measurement->GetChannels()) {
+      for (const auto &s : c.GetSamples()) {
+         for (const auto &norm : s.GetNormFactorList()) {
+            if (!varlist.has_child(norm.GetName())) {
+               auto &v = varlist[norm.GetName()];
+               v.set_map();
+               v["value"] << norm.GetVal();
+               v["min"] << norm.GetLow();
+               v["max"] << norm.GetHigh();
+               if (norm.GetConst()) {
+                  v["const"] << true;
+               }
+            }
+         }
+         for (const auto &sys : s.GetOverallSysList()) {
+            std::string parname("alpha_");
+            parname += sys.GetName();
+            if (!varlist.has_child(parname)) {
+               auto &v = varlist[parname];
+               v.set_map();
+               v["value"] << 0.;
+               v["min"] << -5.;
+               v["max"] << 5.;
+            }
+         }
+      }
+   }
+   for (const auto &sys : this->_measurement->GetConstantParams()) {
+      std::string parname = "alpha_" + sys;
+      if (!varlist.has_child(parname)) {
+         auto &v = varlist[parname];
+         v.set_map();
+      }
+      varlist[parname]["const"] << true;
+   }
+
+   for (const auto &poi : this->_measurement->GetPOIList()) {
+      if (!varlist[poi].has_child("tags")) {
+         auto &tags = varlist[poi]["tags"];
+         tags.set_seq();
+      }
+      varlist[poi]["tags"].append_child() << "poi";
+   }
+
+   // the data
+   auto &datalist = n["data"];
+   datalist.set_map();
+   auto &obsdata = datalist["obsData"];
+   obsdata.set_map();
+   obsdata["index"] << "channelCat";
+   for (const auto &c : this->_measurement->GetChannels()) {
+      std::vector<std::string> obsnames;
+      obsnames.push_back("obs_x_" + c.GetName());
+      obsnames.push_back("obs_y_" + c.GetName());
+      obsnames.push_back("obs_z_" + c.GetName());
+
+      auto &chdata = obsdata[c.GetName()];
+      RooJSONFactoryWSTool::exportHistogram(*c.GetData().GetHisto(), chdata, obsnames);
+   }
+}
+
+void RooStats::HistFactory::JSONTool::PrintJSON(std::ostream &os)
+{
+   tree_t p;
+   auto &n = p.rootnode();
+   n.set_map();
+   this->Export(n);
+   n.writeJSON(os);
+}
+void RooStats::HistFactory::JSONTool::PrintJSON(std::string filename)
+{
+   std::ofstream out(filename);
+   this->PrintJSON(out);
+}
+
+void RooStats::HistFactory::JSONTool::PrintYAML(std::ostream &os)
+{
+#ifdef R__HAS_RYML
+   TRYMLTree p;
+   auto &n = p.rootnode();
+   n.set_map();
+   this->Export(n);
+   n.writeYML(os);
+#else
+   std::cerr << "YAML export only support with rapidyaml!" << std::endl;
+#endif
+}
```

Comment:
```suggestion
#ifdef R__HAS_RYML    
void RooStats::HistFactory::JSONTool::PrintYAML(std::ostream &os)    
{    
   TRYMLTree p;    
   auto &n = p.rootnode();    
   n.set_map();    
   this->Export(n);    
   n.writeYML(os);    
}    
#else    
void RooStats::HistFactory::JSONTool::PrintYAML(std::ostream & /*os*/)       
{    
   std::cerr << "YAML export only support with rapidyaml!" << std::endl;    
}    
#endif    

```
To suppress the warning about the unused `os`, we have to also wrap the function in the signature into the `ifdef` and comment out `os` in the no RYML case.